server/channels/app/platform/websocket_router.go

```diff
@@ -84,7 +84,11 @@ func (wr *WebSocketRouter) ServeWebSocket(conn *WebConn, r *model.WebSocketReque
 		}
 		if thChannelID, ok := r.Data["thread_channel_id"].(string); ok {
 			// Set the channelID of the active thread.
-			conn.SetActiveThreadChannelID(thChannelID)
+			if isThreadView, ok := r.Data["is_thread_view"].(bool); ok && isThreadView {
+				conn.SetActiveThreadViewThreadChannelID(thChannelID)
```

Comment:
Can the active thread view be unset?

Reply:
You mean as in set to empty? Yes, when the user moves out of the global threads view, then it's unset.

Reply:
So in practice to unset it a client would have to send an empty `thread_channel_id` value and a true `is_thread_view`, is that correct?

Reply:
Correct, which it does, when it moves out of thread view.

I am going to optimize some of this in https://github.com/mattermost/mattermost/pull/25723. This whole "<>" business is to handle mobile clients which may not set the scope, but still need to work. But as soon as we get any one value either from active team/channel/thread, then we can set all of them to "". Otherwise, the client needs to open atleast one RHS thread, and one main thread for the scoping to be activated. I will add a `sync.Once` in the webconn to optimize this better. More to come in that PR.